File: src/com/demo/leetcode/study/easy/EasyAri3.java
package com.demo.leetcode.study.easy;

import java.util.Arrays;

public class EasyAri3 {
    public void rotate(int[] nums, int k) {
        k = k % nums.length;
        int[] rightpart = Arrays.copyOfRange(nums, nums.length - k, nums.length);
        System.arraycopy(nums, 0, nums, k, nums.length - k);
        System.arraycopy(rightpart, 0, nums, 0, k);
    }

}

File: src/com/demo/leetcode/study/easy/EasyAri3.kt
package com.demo.leetcode.study.easy

import java.util.*

fun main() {
    rotate(intArrayOf(1,2), 3)
//    rotate(intArrayOf(1, 2, 3, 4, 5, 6, 7), 3)
}

fun rotate(nums: IntArray, k: Int): Unit {
    var nk = k % nums.size
    if (nums.size == 1) {
        return
    }
    var top = nums.sliceArray(0 until nums.size - nk)
    var end = nums.sliceArray(nums.size - nk until nums.size)
    var nNums = end.plus(top)
    for (i in nums.indices) {
        nums[i] = nNums[i]
    }
    System.out.println(Arrays.toString(nums))
}